app/src/main/java/me/jfenn/alarmio/adapters/AlarmsAdapter.kt

```diff
@@ -447,8 +448,7 @@ class AlarmsAdapter(private val alarmio: Alarmio, private val recycler: Recycler
                 }
 
                 override fun afterTextChanged(editable: Editable) {
-                    if (name.hasFocus()) // ignore .setText() calls
-                        alarms.getOrNull(adapterPosition)?.setName(alarmio, editable.toString())
+                    alarms[adapterPosition-timers.size].setName(alarmio, editable.toString())
```

Comment:
Good catch! I definitely missed this problem before.

I would still stick to null-safety here, and the `name.hasFocus()` check is important to prevent the adapter's own `.setText()` calls from invoking it - although it looks like you were initially working off the `master` branch, so you may have missed these changes :)

```suggestion
                    if (name.hasFocus()) // ignore .setText() calls
                        alarms.getOrNull(adapterPosition-timers.size)?.setName(alarmio, editable.toString())
```